java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -65,12 +65,11 @@ public boolean isSubtype(JavaType t, JavaType s) {
           }
 
           //FIXME work on erased types while generics method is not implemented/read from bytecode.
-          Set<JavaType> erasedTypes = Sets.newHashSet();
+          Set<String> erasedTypes = Sets.newHashSet();
           for (JavaType.ClassJavaType classType : t.getSymbol().superTypes()) {
-            erasedTypes.add(classType.erasure());
+            erasedTypes.add(classType.erasure().fullyQualifiedName());
```

Comment:
I might be wrong but I am really wondering if you did not, somehow shortcut erasure by relying on fully qualified name and therefore hacked a solution instead of fixing the real issue... We would need to discuss in depth what is going on here.